compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.cpp

```diff
@@ -384,6 +394,11 @@ FailureOr<GPUMMASchedule> deduceMMASchedule(
       continue;
     }
 
+    // Currently we always do C promotion when the problem doesn't align with
+    // the intrinsic. TODO (jerryyin): Make this a tuning parameter when we are
+    // capable of not doing C promotion in unaligned cases.
+    bool doCPromotion = !mustBeAligned;
```

Comment:
We are already capable of doing c promotion in aligned case, see this test
https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/LLVMGPU/test/ROCDL/pipeline_tile_and_fuse.mlir#L1031

I think what we want is to have `doCPromotion` as an input argument to `deduceMMASchedule` , maybe thats what you meant by the TODO above but it can already be done and we shouldnt try to connect it implicitly with the alignment.

Reply:
Agreed, I double checked the code base for scenarios where we'd `setPromotedOperandList(..., {0,1,2});` and confirmed that this new set you've added is the only place we'd care about. (The other place - set attention vector distribute - is less relevant and has its own heuristic derivation logic).

I've also investigated a bit about the other alternative: not using the `doCPromotion` as an argument carried through the stack when attempting to getting matmul lowering config. But rather, directly query the `promote_operands` field from the `#iree_gpu.lowering_config`. As it turns out, the field is only written after all tuning parameters have been deduced. So ordering wise, this alternative would make as much sense as we just pass `doCPromotion` as an argument as I currently implemented.

With this, I'll rebase and add `doCPromotion` through the calling stack in the following commit.

Reply:
Agreed, I double checked the code base for scenarios where we'd `setPromotedOperandList(..., {0,1,2});` and confirmed that this new set you've added is the only place we'd care about. (The other place - set attention vector distribute - is less relevant and has its own heuristic derivation logic).

I've also investigated a bit about the other alternative: not using the `doCPromotion` as an argument carried through the stack when attempting to getting matmul lowering config. But rather, directly query the `promote_operands` field from the `#iree_gpu.lowering_config`. As it turns out, the field is only written after all tuning parameters have been deduced. So ordering wise, this alternative would make as much sense as we just pass `doCPromotion` as an argument as I currently implemented.

With this, I'll rebase and add `doCPromotion` through the calling stack in the following commit.